<?xml version="1.0" encoding="UTF-8"?>
<!--
Author: Tomas Brablec
Project: pacppman (ICP project)
License: GPL 3.0 (https://www.gnu.org/licenses/gpl-3.0.txt)
Module: MainWindow
Description: This file contatins the XML description of the user interface,
created using QT Designer.
-->
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>551</width>
    <height>455</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>pacppman</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="stepsLabel">
        <property name="text">
         <string>&lt;b style=&quot;color: orange;&quot;&gt; 0 &lt;/b&gt; steps taken</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="text">
         <string>press Space to start</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="timeLabel">
        <property name="text">
         <string>0:00</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="keysLabel">
        <property name="text">
         <string>&lt;b style=&quot;color: orange;&quot;&gt; 0 &lt;/b&gt; keys left</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::DefaultContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">border: none; background-color: black</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>551</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Map"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuRecording">
    <property name="title">
     <string>Recording</string>
    </property>
    <addaction name="actionSave_Recording"/>
    <addaction name="actionOpen_Recording"/>
    <addaction name="separator"/>
    <addaction name="actionPlay"/>
    <addaction name="actionPause"/>
    <addaction name="separator"/>
    <addaction name="actionStep"/>
    <addaction name="actionStep_Back"/>
   </widget>
   <widget class="QMenu" name="menuGame">
    <property name="title">
     <string>Game</string>
    </property>
    <addaction name="actionStart"/>
    <addaction name="actionRestart"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuGame"/>
   <addaction name="menuRecording"/>
  </widget>
  <action name="actionOpen_Map">
   <property name="text">
    <string>Open Map</string>
   </property>
  </action>
  <action name="actionSave_Recording">
   <property name="text">
    <string>Save Recording</string>
   </property>
  </action>
  <action name="actionOpen_Recording">
   <property name="text">
    <string>Open Recording</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="text">
    <string>Step</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionPlay">
   <property name="text">
    <string>Play</string>
   </property>
  </action>
  <action name="actionStep_Back">
   <property name="text">
    <string>Step Back</string>
   </property>
   <property name="shortcut">
    <string>B</string>
   </property>
  </action>
  <action name="actionStart">
   <property name="text">
    <string>Start</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionRestart">
   <property name="text">
    <string>Restart</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="text">
    <string>Pause</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen_Map</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openMap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Recording</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_Recording</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStep</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>stepRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPlay</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>playRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStep_Back</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>stepBackRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStart</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>startGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRestart</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>restartGame()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPause</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>pauseRecording()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>quit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>gamePausedStateChanged(bool)</signal>
  <signal>fileLoadStatusChanged(bool)</signal>
  <slot>openMap()</slot>
  <slot>saveRecording()</slot>
  <slot>openRecording()</slot>
  <slot>playRecording()</slot>
  <slot>stepRecording()</slot>
  <slot>stepBackRecording()</slot>
  <slot>startGame()</slot>
  <slot>restartGame()</slot>
  <slot>pauseRecording()</slot>
  <slot>quit()</slot>
 </slots>
</ui>
